### ISSUE TITLE
Support tests expecting Exceptions

### ISSUE THREAD

**Issue by skinny85** [skinny85/specnaz#2]
In "regular" JUnit, you can write a test like so:

```java
import org.junit.Test;

import java.util.ArrayList;

public class ExpectTest {
    @Test(expected = IndexOutOfBoundsException.class)
    public void expect_exception_test() throws Exception {
        new ArrayList<String>().get(0);
    }
}
```

There was a Feature Request to add a similar capability to Specnaz.

**Comment by skinny85** [comment]
The reason I'm not in love with this feature is that I never liked the JUnit's `expect` attribute. In my opinion, the tests are less readable with it, and a lot more fragile (I've often seen tests that passed, but the actual Exception was not thrown from the line that the author intended, and so they were in fact 'false positives').

As long as Specnaz doesn't have this feature, it forces people to look for alternatives - which is great, because a lot of alternatives exist, and they are all better than JUnit's `expect`:
* If you're using [AssertJ](http://joel-costigliola.github.io/assertj/assertj-core.html) for assertions, you have the [assertThatThrownBy](http://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#exception-assertion) method for that exact scenario.
* There is the [CatchException](https://github.com/Codearte/catch-exception/tree/2.x) library, tailor-built for this purpose.
* Probably many others that I'm forgetting right now.

Design-wise, I don't see a problem with this feature. I would probably make the API look something like:
```java
it.shoudThrow(IllegalArgumentException.class, "when called with a null argument", () -> {
    // the test name will be: "should throw IllegalArgumentException when called with a null argument"
    // ...
});
```

The perfect API would probably be:

```java
it.shouldThrow<IllegalArgumentException>("when called with a null argument", () -> {
    // ...
});
```
..but I'm not sure that's possible with Java. I do know that [KotlinTest does something like this](https://github.com/kotlintest/kotlintest/blob/master/doc/reference.md#exceptions-), so perhaps that will be the API in the Kotlin version.

The only question is: do we want to do this, or do we rather skip this, forcing people to look for (better) alternatives?

**Comment by skinny85** [comment]
Fixed in 1.1 - resolving.
